pylintrc_tweaks

```diff
@@ -34,6 +34,32 @@ disable+ =
     no-name-in-module,
     unnecessary-lambda-assignment,
 
+    unused-import,
+    yield-outside-function,
+    return-outside-function,
+    line-too-long,
+    bare-except,
+    syntax-error,
+    trailing-whitespace,
+    empty-docstring,
+    unnecessary-pass,
+    return-in-init,
+    continue-in-finally,
+    duplicate-bases,
+    unexpected-special-method-signature,
+    invalid-all-object,
+    misplaced-bare-raise,
+    logging-too-many-args,
+    logging-too-few-args,
+    bad-str-strip-call,
+    too-many-statements,
+    too-many-boolean-expressions,
+    useless-else-on-loop,
+    import-self,
+    global-variable-not-assigned,
+    global-at-module-level,
+    binary-op-exception,
```

Comment:
According to https://github.com/astral-sh/ruff/issues/970 , the following additional checks which are enabled in our pylintrc are also implemented in ruff.

anomalous-backslash-in-string
assert-on-tuple
bad-classmethod-argument
bad-format-string
bad-format-string-key
bad-open-mode
boolean-datetime
cell-var-from-loop
consider-iterating-dictionary
dangerous-default-value
duplicate-argument-name
duplicate-except
duplicate-key
eval-used
exec-used
expression-not-assigned
format-combined-specification
format-needs-mapping
function-redefined
logging-format-interpolation
logging-not-lazy
lost-exception
misplaced-future
missing-final-newline
missing-format-argument-key
missing-format-string-key
multiple-imports
multiple-statements
no-method-argument
no-self-argument
nonexistent-operator
not-in-loop
pointless-statement
redefined-builtin
reimported
return-arg-in-generator
singleton-comparison
too-few-format-args
too-many-format-args
truncated-format-string
undefined-all-variable
undefined-variable
unidiomatic-typecheck
unnecessary-lambda
unnecessary-semicolon
unneeded-not
unused-argument
unused-format-string-argument
unused-format-string-key
unused-variable
wildcard-import
wrong-import-order
wrong-import-position